infra/graph_generators.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Mar 17 13:00:33 2019
+
+@author: Zargham
+"""
+import networkx as nx
+
+def lineGraphGen(N, bidir=False,nodeTypeName='vanilla',edgeTypeName='vanilla'):
```

Comment:
nit, all of these public methods should have python docstrings:
https://www.python.org/dev/peps/pep-0257/

Reply:
Added doc string; not sure they are totally up to convention but i read the link and added simple statements about each function.